src/notebooks/helpers.ts

```diff
@@ -912,7 +912,7 @@ export function updateNotebookMetadata(
     if (kernelConnection?.kind === 'startUsingPythonInterpreter') {
         // Store interpreter name, we expect the kernel finder will find the corresponding interpreter based on this name.
         const kernelSpec = kernelConnection.kernelSpec;
-        let displayName = metadata.kernelspec?.display_name || kernelConnection.interpreter.displayName || '';
+        let displayName = kernelSpec.display_name || kernelConnection.interpreter.displayName || '';
```

Comment:
A different semi-related bug. After making the initial fix the display_name was not getting updated correctly. Here it's pulling from the old saved display name instead of the new display name that we are switching to.